iceoryx2-ffi/ffi/src/api/file_descriptor.rs

```diff
@@ -0,0 +1,200 @@
+// Copyright (c) 2024 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Apache Software License 2.0 which is available at
+// https://www.apache.org/licenses/LICENSE-2.0, or the MIT license
+// which is available at https://opensource.org/licenses/MIT.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+#![allow(non_camel_case_types)]
+
+use iceoryx2_bb_elementary::static_assert::*;
+use iceoryx2_bb_posix::{
+    file_descriptor::{FileDescriptor, FileDescriptorBased},
+    file_descriptor_set::SynchronousMultiplexing,
+};
+use iceoryx2_ffi_macros::iceoryx2_ffi;
+
+use super::{AssertNonNullHandle, HandleToType};
+
+// BEGIN type definition
+
+#[derive(Debug)]
+#[repr(C)]
+pub struct CFileDescriptor {
+    value: i32,
+    is_owned: bool,
+}
+
+impl FileDescriptorBased for CFileDescriptor {
+    fn file_descriptor(&self) -> &FileDescriptor {
+        unsafe { core::mem::transmute(self) }
+    }
+}
+
+impl SynchronousMultiplexing for CFileDescriptor {}
```

Comment:
This is superfluous and can be removed. Just change
```rs
pub type iox2_file_descriptor_ptr = *const CFileDescriptor;
pub type iox2_file_descriptor_ptr_mut = *mut CFileDescriptor;
```
to
```rs
pub type iox2_file_descriptor_ptr = *const FileDescriptor;
pub type iox2_file_descriptor_ptr_mut = *mut FileDescriptor;
```
and every other usage of `CFileDescriptor` to `FileDescriptor` and it should still work without duplicating the type.

Reply:
Nope, the FileDescriptor itself does not implement SynchronousMultiplexing therefore I added a CFileDescriptor that supports it - in an unsafe manner.